### PR TITLE
Knowing in which organization an assignment is submitted

### PR DESCRIPTION
**Pull request by felipecalvo**
Adds reference so we can keep track of each new assignment's organization.

Old assigmnents, in the meantime, remain with no organization

### CODE REVIEW

File: app/models/assignment.rb [hunk 1]
```diff
@@ -126,7 +127,7 @@ def run_tests!(params)
   end
 
   def to_resource_h
-    as_json(except: [:exercise_id, :submission_id, :id, :submitter_id, :solution, :created_at, :updated_at, :submission_status],
+    as_json(except: [:exercise_id, :submission_id, :organization_id, :id, :submitter_id, :solution, :created_at, :updated_at, :submission_status],
```
Comment (flbulgarelli) [inline app/models/assignment.rb]: I think we should expose the organization name in this resource hash

Comment (felipecalvo) [reply app/models/assignment.rb]: The organization name is exposed on line 141, isn't it?

Comment (flbulgarelli) [reply app/models/assignment.rb]: You are right. Dismiss my comment

File: app/models/assignment.rb [hunk 2]
```diff
@@ -12,6 +12,7 @@ class Assignment < ApplicationRecord
            primary_key: :submission_id,
            dependent: :destroy
 
+  belongs_to :organization, default: -> { Organization.current }
```
Comment (flbulgarelli) [inline app/models/assignment.rb]: Will the organization be updated if it it is resubmitted from another organization?

Comment (felipecalvo) [reply app/models/assignment.rb]: This is changed now and it doesn't depend on default values anymore.

File: app/models/concerns/with_assignments.rb [hunk 3]
```diff
@@ -18,15 +18,15 @@ def has_messages_for?(user)
     messages_for(user).present?
   end
 
-  def find_assignment_for(user)
+  def find_assignment_for(user, organization)
```
Comment (flbulgarelli) [inline app/models/concerns/with_assignments.rb]: :see_no_evil: a dangling `organization` is still here

Comment (flbulgarelli) [reply app/models/concerns/with_assignments.rb]: This is on purpose, right?

If that is the case, please put a comment about it, and name it `_organization`

File: lib/mumuki/domain/factories/organization_factory.rb [hunk 4]
```diff
@@ -21,7 +19,20 @@
...
+    name { 'test' }
+    book { create(:book, name: 'test', slug: 'mumuki/mumuki-the-book') }
+  end
+
+  factory :another_test_organization, parent: :test_organization, traits: [:skip_unique_name_validation] do
+    book { create(:book, name: 'another-test', slug: 'mumuki/mumuki-another-book') }
+  end
+
+  trait :skip_unique_name_validation do
+    to_create { |instance| instance.save(validate: false) }
```
Comment (flbulgarelli) [inline lib/mumuki/domain/factories/organization_factory.rb]: Didn't know `to_create` :clap:

File: app/models/concerns/with_assignments.rb (hunk 3 above)
Comment (felipecalvo) [reply app/models/concerns/with_assignments.rb]: Yes, it's all ready to filter by organization by changing
`assignments.find_by(submitter: user)`
to
`assignments.find_by(submitter: user, organization: organization)`
when [the time comes](https://github.com/mumuki/ikumi-online/issues/464).

Comment (felipecalvo) [reply app/models/concerns/with_assignments.rb]: I'll change it to `_organization`, didn't know about the convention

**Comment by flbulgarelli** [comment]
@felipecalvo this looks good to me. Please merge if ready.

**Pull request merged**